hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/BatchConsumer.java

```diff
@@ -0,0 +1,168 @@
+    }
+
+    @Override
+    public void run() {
+        setThreadName();
+        while (isConsuming()) {
+            Optional<MessageBatch> inflight = Optional.empty();
+            try {
+                logger.debug("Trying to create new batch for subscription {}", subscription.getId());
+                MessageBatchingResult result = receiver.next(subscription);
+                inflight = of(result.getBatch());
+                inflight.ifPresent(batch -> {
+                    logger.debug("Delivering batch for subscription {}", subscription.getId());
+                    deliver(batch, createRetryer(batch, subscription.getBatchSubscriptionPolicy()));
+                    logger.debug("Finished delivering batch for subscription {}", subscription.getId());
+                    offsets.putAllDelivered(batch.getPartitionOffsets());
+                });
+                result.getDiscarded().forEach(m -> monitoring.markDiscarded(m, subscription, "too large"));
+            } finally {
+                logger.debug("Cleaning batch for subscription {}", subscription.getId());
+                inflight.ifPresent(this::clean);
+            }
+        }
+        logger.info("Stopped consumer for subscription {}", subscription.getId());
+        unsetThreadName();
+        stoppedLatch.countDown();
+    }
+
+    private Retryer<MessageSendingResult> createRetryer(MessageBatch batch, BatchSubscriptionPolicy policy) {
+        return createRetryer(batch, policy.getMessageBackoff(), policy.getMessageTtl(), policy.isRetryClientErrors());
+    }
+
+    private Retryer<MessageSendingResult> createRetryer(final MessageBatch batch, int messageBackoff, int messageTtl, boolean retryClientErrors) {
+        return RetryerBuilder.<MessageSendingResult>newBuilder()
+                .retryIfExceptionOfType(IOException.class)
+                .retryIfRuntimeException()
+                .retryIfResult(result -> isConsuming() && !result.succeeded() && (!result.isClientError() || retryClientErrors))
+                .withWaitStrategy(fixedWait(messageBackoff, MILLISECONDS))
+                .withStopStrategy(attempt -> attempt.getDelaySinceFirstAttempt() > messageTtl)
+                .withRetryListener(getRetryListener(result -> monitoring.markFailed(batch, subscription, result)))
+                .build();
+    }
+
+    private void deliver(MessageBatch batch, Retryer<MessageSendingResult> retryer) {
+        try (Timer.Context timer = hermesMetrics.subscriptionLatencyTimer(subscription).time()) {
+            MessageSendingResult result = retryer.call(() -> sender.send(batch, subscription.getEndpoint(), subscription.getBatchSubscriptionPolicy().getRequestTimeout()));
+            monitoring.markSendingResult(batch, subscription, result);
+        } catch (Exception e) {
+            logger.error(format("[batch_id=%s, subscription=%s] Batch was rejected.", batch.getId(), subscription.toSubscriptionName()), e);
```

Comment:
another log format? maybe make it compatible with existing? (or change existing)